dashboard/app/models/video.rb

```diff
@@ -24,6 +24,7 @@ class Video < ActiveRecord::Base
   scope :current_locale, -> {where(locale: I18n.locale.to_s).or(Video.english_locale).unscope(:order).order("(case when locale = 'en-US' then 0 else 1 end) desc")}
 
   validates_uniqueness_of :key, scope: [:locale]
+  validates :key, format: {with: /\A(?~:)\z/}
```

Comment:
Is a colon is the only invalid character?

Reply:
Good question!  I used several examples of video keys currently in use and tested against regex.  Video keys can have combinations of letters (upper & lower case), numbers and underscore.
Some examples tested:
maze_intro
C1_bigevent_unplugged
unplug11
mc_2017_functions
20_wrapup

Reply:
Good call!  /\A(?~:)\z/ - already allows for uppercase, lowercase, numbers and underscores.  The regex expression /\A[a-zA-Z0-9_]+\z/ is more explicit if we are sure that we only allow letters, numbers and underscores.

Here is an awesome tool I use to craft regex that I like https://rubular.com/

Reply:
A number of keys use dashes as well, so we might want to add those in too (ex CSF_bb8_skinOverview_C-F).

I'm in favor of a more restrictive regex though. I'm skeptical that a colon is the only character we should be checking for.

Reply:
Good call, I missed that in my quick scan of videos.csv! So maybe:
```rb
/\A[a-zA-Z0-9\-_]+\z/
```